### DISCUSSION TITLE
Updating one field's value from another field's component

### DISCUSSION THREAD

**Discussion by ninahorne** [payloadcms/payload#486] [Q&A]
I am building out some custom components that copy data from other fields. I have been using `useWatchForm` and getting mostly the desired behavior. I can change the other field's form value, but the UI for that field doesn't update to reflect the new data. For most fields, this is no problem and I can create a custom component that will listen to changes and update. I got stuck while trying to do this for a RichText component. It looks like Payload doesn't expose that component. 

Is there a way to trigger the form to update its values based on the form's current values?
Or would it be possible to expose the RichText editor so that I could use it in a custom component?

Any help would be greatly appreciated!

Edit:

As I work with things further, I think I would prefer to have the ability to trigger the form to update its inputs based on update field values. That way field types like 'array' can be easily updated from another component without having to deal with the complexity of building a custom component.

**Comment by jmikrut** [comment]
Hey @ninahorne — sorry for delay in response here. We have been working on brainstorming ways to solve this exact problem. 

It is the way it is right now for performance reasons, but we do have a potential idea for how to get around this. We're working on a proof of concept that will do exactly what you're asking for - have a field update in response to its value updating, even if that update comes programmatically from another field entirely. 

In the meantime, we can certainly expose the `RichText` component for use within custom components and you can tie into it the same way you've done elsewhere. We'll slate this into our workflow and report back here once we have that component exportable.

We'll also continue down this PoC that I mentioned above. That'll be the best way to solve this for sure.

Will comment here with further updates shortly!

**Reply by ninahorne** [reply]
Thanks for the response, James! That all makes a lot of sense. Just an FYI, I decided to go a different route with my project so no need to rush on this feature on my account!!
